lib/private/RedisFactory.php

```
@@ -38,7 +41,12 @@ public function __construct(SystemConfig $config) {
 		$this->config = $config;
 	}
 
+	/**
+	 * @throws \Exception
```

should throw `UnexpectedValueException`